localization/ndt_scan_matcher/src/map_update_module.cpp

```diff
@@ -146,12 +146,26 @@ bool MapUpdateModule::update_ndt(const geometry_msgs::msg::Point & position, Ndt
     request,
     [](rclcpp::Client<autoware_map_msgs::srv::GetDifferentialPointCloudMap>::SharedFuture) {})};
 
+  // Wait for maximum 10 milliseconds
+  std::chrono::milliseconds timeout(10);
+  auto start = std::chrono::system_clock::now();
+
   std::future_status status = result.wait_for(std::chrono::seconds(0));
   while (status != std::future_status::ready) {
     RCLCPP_INFO(logger_, "waiting response");
     if (!rclcpp::ok()) {
       return false;  // No update
     }
+
+    auto cur = std::chrono::system_clock::now();
+
+    // Report an error if wait for too long
+    if (cur - start >= timeout) {
+      RCLCPP_ERROR_STREAM_THROTTLE(
+        logger_, *clock_, 1000, "Waited for incoming PCDs for too long. Abandon NDT update.");
+      return false;
```

Comment:
```suggestion
```

Please remove this line because if it is returned, it will cause the program to stop running.

Reply:
@YamatoAndo So even when no PCD coming at all, the matching still continues? That does not sound right to me. If the map is not available then the matching should drop and a re-initialization should be performed.